subsys/spm/secure_services.c

```diff
@@ -41,6 +42,10 @@
 #include <mbedtls/entropy_poll.h>
 #endif /* CONFIG_SPM_SERVICE_RNG */
 
+static bool ptr_in_secure_ram(intptr_t ptr)
+{
+	return ptr < (CONFIG_SRAM_BASE_ADDRESS + NON_SECURE_RAM_OFFSET)
```

Comment:
We should have PM symbols for this SPM_SRAM or SRAM_SECURE.

Reply:
We do, but we are still supporting users which does not use multi image build, and hence not PM either.

Reply:
Ok, I see. I think we need to fix the logic in spm_internal.h to be future proof in case we rearrange the RAM sections. We can do that now that we have RAM partitioning in PM.

Reply:
Rearrange - in what way? If the secure ram is no longer starting at the start of RAM? Note that we must be compatible with the devicetree symbols as well.

Reply:
Ok, we don't need to get into it here.

One thing we could do in this PR though is

    #if CONFIG_PM_SRAM_SECURE_OFFSET != 0x20000000
        #error Secure RAM must start at the beginning of RAM when using SPM.
    #endif

It's up to you.

Reply:
Resolving this since we will use the cmse function instead :+1: